grpc/src/test/java/com/linecorp/armeria/server/grpc/GrpcServerInterceptorTest.java

```diff
@@ -131,4 +162,67 @@ public <I, O> Listener<I> interceptCall(ServerCall<I, O> call, Metadata metadata
             return next.startCall(call, metadata);
         }
     }
+
+    private static final Metadata.Key<String> TEST_HEADER = Metadata.Key.of("test",
+                                                                            Metadata.ASCII_STRING_MARSHALLER);
+
+    private static <REQ, RESP> ServerCall<REQ, RESP> addToHeader(String suffix, ServerCall<REQ, RESP> call) {
+        return new SimpleForwardingServerCall<REQ, RESP>(call) {
+            @Override
+            public void sendHeaders(Metadata responseHeaders) {
+                final String testHeader;
+                if (responseHeaders.get(TEST_HEADER) != null) {
+                    testHeader = responseHeaders.get(TEST_HEADER);
+                } else {
+                    testHeader = "";
+                }
+
+                responseHeaders.put(TEST_HEADER, testHeader + suffix);
+                super.sendHeaders(responseHeaders);
+            }
+        };
+    }
+
+    private static class InterceptorAddFoo implements ServerInterceptor {
+        @Override
+        public <REQ, RESP> Listener<REQ> interceptCall(ServerCall<REQ, RESP> call, Metadata headers,
+                                                       ServerCallHandler<REQ, RESP> next) {
+            return next.startCall(addToHeader("foo", call), headers);
+        }
+    }
+
+    private static class InterceptorAddBar implements ServerInterceptor {
+        @Override
+        public <REQ, RESP> Listener<REQ> interceptCall(ServerCall<REQ, RESP> call, Metadata headers,
+                                                       ServerCallHandler<REQ, RESP> next) {
+
+            return next.startCall(addToHeader("bar", call), headers);
+        }
+    }
+
+    private static class InterceptorAddQux implements ServerInterceptor {
+        @Override
+        public <REQ, RESP> Listener<REQ> interceptCall(ServerCall<REQ, RESP> call, Metadata headers,
+                                                       ServerCallHandler<REQ, RESP> next) {
+
+            return next.startCall(addToHeader("qux", call), headers);
+        }
+    }
+
+    @GrpcInterceptor(InterceptorAddBar.class)
+    private static class AnnotatedInterceptorServiceImpl extends TestServiceImplBase {
+
+        @Override
+        @GrpcInterceptor(InterceptorAddQux.class)
```

Comment:
I think the order should be:
- global decorator -> class level decorator -> method level decorator
like how `@Decorator` works.

However, the current order is reversed. Could you fix it?

Reply:
Good point, let me fix 👍

Reply:
Hmm taking a look at the test,

```
assertThat(headersCapture.get().get(TEST_HEADER)).isEqualTo("foobarqux");
```

Here first `Foo` (global) is appended, later `bar` which is the class level and finally `qux` which is the method level. Am I reading this incorrectly? I feel like this is the right order?

Reply:
Note that we append headers in requests, not responses so the order of  `foo bar qux` indicate order of decorators being executed.

Reply:
Fixed 🙌